app/src/main/java/com/pnow/rick_and_morty_list/app/ui/viewmodel/DetailsViewModel.kt

```diff
@@ -21,40 +26,81 @@ class DetailsViewModel @Inject constructor(
     private val rickAndMortyRepository: RickAndMortyRepository,
     private val detailsMapper: DetailsUiMapper,
     private val dispatcher: CoroutineDispatcher
-) :
-    ViewModel() {
+) : ViewModel() {
 
-    private val _detailsState = MutableLiveData<DetailsUIModel>()
-    val detailsState: LiveData<DetailsUIModel> = _detailsState
+    private val _detailsState =
+        MutableStateFlow<DetailsUiState>(DetailsUiState.CharacterInfoLoading)
+    val detailsState: StateFlow<DetailsUiState> = _detailsState.asStateFlow()
 
-    fun fetchDetails(episodesUrlList: List<String>?, locationUrl: String?, originUrl: String?) {
+    companion object {
+        const val TAG = "DetailsVM"
```

Comment:
Sprawdz sobie Timber do logów. Można go mega spoko skonfigurować i porobić hyperliniki w logcat. Klikasz i Cie przenosi w to miejsce w kodzie